src/excmds.ts

```diff
@@ -961,6 +976,168 @@ export function jumpprev(n = 1) {
     })
 }
 
+/**
+ * Jumps to a local mark, a global mark, or the location before the last mark jump.
+ * [a-z] are local marks, [A-Z] are global marks and '`' is the location before the last mark jump.
+ * @param key the key associated with the mark
+ */
+//#content
+export async function jumpmark(key: string) {
+    if (key.length !== 1) {
+        return;
+    }
+    if (key === "`") {
+        return jumpBeforeJumpMark()
+    }
+    if (!/[a-z]/i.exec(key)) {
+        return;
+    }
+    if (key === key.toUpperCase()) {
+        return jumpGlobalMark(key);
+    }
+    return jumpLocalMark(key);
+}
+
+/**
+ * Jumps to a local mark.
+ * @param key the key associated with the mark
+ * @hidden
+ */
+async function jumpLocalMark(key: string) {
+    const urlWithoutAnchor = window.location.href.split("#")[0]
+    const localMarks = await State.getAsync("localMarks");
+    const mark = localMarks.get(urlWithoutAnchor)?.get(key);
+    if (mark) {
+        const currentTabId = await activeTabId();
+        state.beforeJumpMark = { scrollX: window.scrollX, scrollY: window.scrollY, tabId: currentTabId };
+        window.scrollTo(mark.scrollX, mark.scrollY);
+        return fillcmdline_tmp(3000, `Jumped to mark '${key}'`);
+    }
+    return fillcmdline_tmp(3000, `Mark '${key}' is not set`);
+}
+
+/**
+ * Jumps to a global mark. If the tab with the mark no longer
+ * exists, jumps to another tab with the same url or creates it first if such tab does not exist.
+ * @param key the key associated with the mark
+ * @hidden
+ */
+async function jumpGlobalMark(key: string) {
+    const globalMarks = await State.getAsync("globalMarks");
+    const mark = globalMarks.get(key);
+    if (!mark) {
+        return fillcmdline_tmp(3000, `Mark '${key}' is not set`);
+    }
+    const currentTabId = await activeTabId();
+    state.beforeJumpMark = { scrollX: window.scrollX, scrollY: window.scrollY, tabId: currentTabId };
+    try {
+        const tab = await getTabByTabId(mark.tabId);
+        goToTab(tab.id).then(() => {
```

Comment:
```
Messaging.messageTab(tab.id, "controller_content", "acceptExCmd", [`composite scrollto ${mark.scrollX} ${mark.scrollY}; fillcmdline_tmp 3000 Jumped to mark '${key}'`])
```

I think this is how I'd solve your problem. We should probably make a little `acceptExCmd(destination: tabId | "background", ex_string)` helper function so we don't have to remember this strange incantation every time.

Reply:
Thanks that works but when opening a new tab from a global mark the message still appeared in the old tab for me. I made the scroll tab like this: `browserBg.tabs.executeScript(//scrolling).then(//printing) which seems to make the message appear in a new tab for me most of the time.